utils/authorization.go

```diff
@@ -28,13 +28,17 @@ func SetRolePermissionsFromConfig(roles map[string]*model.Role, cfg *model.Confi
 		)
 	}
 
-	if IsLicensed() {
+	if isLicensed {
 		switch *cfg.TeamSettings.RestrictPublicChannelManagement {
 		case model.PERMISSIONS_ALL:
 			roles[model.TEAM_USER_ROLE_ID].Permissions = append(
 				roles[model.TEAM_USER_ROLE_ID].Permissions,
 				model.PERMISSION_MANAGE_PUBLIC_CHANNEL_PROPERTIES.Id,
 			)
+			roles[model.CHANNEL_USER_ROLE_ID].Permissions = append(
```

Comment:
The `isLicensed == true && PERMISSIONS_ALL` case should set the exact same permissions as the `isLicensed == false` case, so this addition also needs to be made in the else block.

Reply:
Also, I'm wondering if we should *remove* these four cases from `TEAM_USER` as well. Or might there be some customer scenario that would be messed up by us doing that?

Reply:
@grundleborg Added permissions to non-licensed cases too. All tests seem green without the team_user role having those permissions so its all good from a config-mapping POV.